### PR TITLE
Merge master 2017 01 04

### PR DESCRIPTION
**Pull request by noirbizarre**

### CODE REVIEW

File: js/locales/udata.en.json [hunk 1]
```diff
@@ -123,6 +123,13 @@
     "Drag a picture here": "Drag a picture here",
     "Drop resource": "Drop resource",
     "Edit": "Edit",
+    "Edit dataset \"{title}\"": "Edit dataset \"{title}\"",
```
Comment (vinyll) [inline js/locales/udata.en.json]: I would use single quotes for the outside text in this context for avoiding the escaping.

File: udata/tests/api/test_datasets_api.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 from datetime import datetime
 
 from flask import url_for
+from werkzeug.datastructures import FileStorage
```
Comment (vinyll) [inline udata/tests/api/test_datasets_api.py]: `FileStorage` seems not to be used in the file

File: udata/tests/api/test_url_api.py [hunk 3]
```diff
@@ -1,42 +1,123 @@
...
 
-from . import APITestCase
 
+from udata.tests.api import APITestCase
+from udata.utils import faker
+
+from udata.settings import Testing
+
+CROQUEMORT_URL = 'http://check.test'
+CHECK_ONE_URL = CROQUEMORT_URL + '/check/one'
```
Comment (vinyll) [inline udata/tests/api/test_url_api.py]: I would `'{}./check/one'.format(CROQUEMORT_URL)`

File: udata/tests/api/test_url_api.py [hunk 4]
```diff
@@ -1,42 +1,123 @@
...
+        response.update(data)
+    return json.dumps(response)
+
+
+def mock_url_check(url, data=None, status=200):
+    url_hash = faker.md5()
+    httpretty.register_uri(httpretty.POST, CHECK_ONE_URL,
+                           body=json.dumps({'url-hash': url_hash}),
+                           content_type='application/json')
+    httpretty.register_uri(httpretty.GET, METADATA_URL + '/' + url_hash,
```
Comment (vinyll) [inline udata/tests/api/test_url_api.py]: I would `'{}/{}'.format(METADATA_URL, url_hash)`

File: udata/tests/api/test_url_api.py [hunk 5]
```diff
@@ -1,42 +1,123 @@
...
+    settings = CheckUrlSettings
+
+    @httpretty.activate
     def test_returned_metadata(self):
-        url = 'https://www.data.gouv.fr/fr/'
-        response = self.get(url_for('api.checkurl'), qs={'url': url})
+        url = faker.uri()
+        metadata = {
+            'content-type': 'text/html; charset=utf-8',
+            'status': '200',
```
Comment (vinyll) [inline udata/tests/api/test_url_api.py]: _str_ vs _int_

File: js/locales/udata.en.json (hunk 1 above)
Comment (noirbizarre) [inline js/locales/udata.en.json]: This is an automatically generated/extracted file

**Pull request merged**